server/src/controllers/user.ts

```diff
@@ -3,99 +3,108 @@
 // SPDX-License-Identifier: MIT
 
 import { Request, Response } from 'express';
-import { Course } from './course';
+import models from '../database/models';
+import Course from '../database/models/course';
+import CourseInstance from '../database/models/courseInstance';
+import CourseTranslation from '../database/models/courseTranslation';
+import { CourseData, Language } from './course';
 
 export interface TeacherCourses {
-  current: Array<Course>,
-  previous: Array<Course>
+  current: Array<CourseData>,
+  previous: Array<CourseData>
 }
 
-// eslint-disable-next-line @typescript-eslint/no-unused-vars
-async function getTeacherCourses(userId: number): Promise<TeacherCourses> {
-  return {
-    current: [
-      {
-        id: 0,
-        courseCode: 'CS-E4580',
-        minCredits: 5,
-        maxCredits: 5,
-        department: {
-          fi: '',
-          sv: '',
-          en: 'Department of Computer Science'
-        },
-        name: {
-          fi: '',
-          sv: '',
-          en: 'Programming Parallel Computers D'
-        },
-        evaluationInformation: {
-          fi: '',
-          sv: '',
-          en: ''
-        }
+export async function getUserCourses(req: Request, res: Response): Promise<void> {
+  try {
+    const teacherCourses: TeacherCourses = { current: [], previous: [] };
+
+    // TODO: Go through course_role instead
+    let instances: Array<CourseInstance> = await models.CourseInstance.findAll({
+      where: {
+        responsibleTeacher: Number(req.params.userId)
```

Comment:
This is just a note for the future. It is better not to put request params directly to the query, rather first run validation on it.